src/main/java/kafdrop/service/KafkaHighLevelAdminClient.java

```diff
@@ -133,6 +133,25 @@ void createTopic(NewTopic newTopic) {
     }
   }
 
+  /**
+   * Delete topic or throw ${@code KafkaAdminClientException}
+   *
+   * @param topic name of the topic to delete
+   * @throws KafkaAdminClientException if computation threw an Exception
+   */
+  void deleteTopic(String topic) {
+    DeleteTopicsOptions options = new DeleteTopicsOptions();
+    options.timeoutMs(5000); // timeout after 5 seconds
+    final var deleteTopicsResult = adminClient.deleteTopics(List.of(topic), options);
+    try {
+      deleteTopicsResult.all().get();
+      LOG.info("Topic {} successfully deleted", topic);
```

Comment:
Please see #97 . I believe #97 's implementation would log the topic name and topic partition number successfully deleted when the future is completed. It could also be overloaded to accept a collection of topic names.

Reply:
I'll check it out, thanks

Reply:
having checked #97 I don't believe this is the case.
`deletionResult.values()` returns a map with key `String` and values `KafkaFuture<Void>`.
Your implementation would take the keyset (String) and print these values.
But with that implementation you're not checking if the futures have completed or not (they are asynchronous).

Reply:
Thanks, I will change #97 to invoke `KafkaFuture.get() ` so that `void deleteTopic(String topic) ` is async. My point was that logging the topic name and partition id is more effective than logging the string passed to `void deleteTopic(String topic) `

Reply:
can you point me in the direction where #97 logs the partition id?
I only see it logging the topic name.
Also don't see the added value in highlighting that specific partitions are deleted.
When deleting a topic you also delete all of its partitions

Reply:
I misread this, logging the ids of successfully deleted partitions would be useful in if a timeout were to occur or for that ensuring replicated topics are deleted across a multi broker cluster. I think this is a good implentation: 

`private void tryDelete(AdminClient adminClient, String topic)
    throws Exception {
  try {
    adminClient.deleteTopics(Collections.singleton(topic)).all().get(DELETE_TIMEOUT_SECONDS, TimeUnit.SECONDS);
  } catch (TimeoutException e) {
    LOG.info("Did not receive delete topic response within %d seconds. Checking if it succeeded",
      DELETE_TIMEOUT_SECONDS);
    if (adminClient.listTopics().names().get(DELETE_TIMEOUT_SECONDS, TimeUnit.SECONDS).contains(topic)) {
      throw new Exception("Topic still exists after timeout");
    }
  }
}
`
See: 

https://github.com/apache/flink/blob/master/flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTestEnvironmentImpl.java#L168